addons/account/static/src/components/x2many_buttons/x2many_buttons.xml

```diff
@@ -2,16 +2,38 @@
 <templates>
     <t t-name="account.X2ManyButtons">
         <div class="d-flex align-items-center gap-1">
-            <t t-foreach="this.currentField.records.slice(0, 3)" t-as="record" t-key="record.id">
-                <button class="btn btn-link p-0"
-                        t-on-click="() => this.openFormAndDiscard(record.resId)"
-                        t-att-data-hotkey="`shift+${record_index + 1}`"
-                        t-out="record.data.display_name"/>
-                <span t-if="!record_last">,</span>
+            <t t-if="this.currentField.resModel == 'account.payment'">
```

Comment:
Here I added a template and then conditionally display it. Is that ok, or would it be better to extend the x2many button class?

Reply:
i think that extending the x2many button class, and override the template will be cleaner :smile: